xc7/utils/vivado_output_timing.py

```diff
@@ -0,0 +1,32 @@
+""" Utility for generating TCL script to output timing information from a
+design checkpoint.
+"""
+import argparse
+
+
+def create_runme(f_out, args):
+    print(
+        """
+report_timing_summary
+
+source {util_tcl}
+write_timing_info timing_{name}.json5
+""".format(name=args.name, util_tcl=args.util_tcl),
+        file=f_out
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument('--name', required=True)
+    parser.add_argument('--util_tcl', required=True)
```

Comment:
So I don't understand the question?  Are you ask what the argument is for?

Reply:
Sorry for the ramble, but trying to think of a simpler approach.

I see that timing_utils.tcl script is being passed, but wonder if we would ever have a reason to pass a different script as the argument. If not, may be they shouldn't be separated into different files. The flow is fragmented across tcl, python, and the cmake invocation.

Maybe there isn't a way around it for generating/running tcl wrapped in python, but this python generates a new tcl script that runs 3 commands and I don't see  using it with different parameters.

Reply:
The purpose of the argument is *not* to enable switching of scripts.  The reason for the argument is two fold.  The first is to avoid knowledge of paths in the templating script.  The second is more fundamental, which is that non-trival TCL should not be located within a python template.

As for whether we can avoid the python script altogether, I think this approach (writing the tcl script and then invoking vivado to source it) is less bad than something that involves passing arguments are environment variables, or having CMake cat the script to a file, etc.

Reply:
ok, sounds good. I figured it might be a matter of some trade-off, I think the path makes sense. I could go either way on if non-trivial TCL is better as separate file or as a string in a template. First option is easier to develop but a little more difficult for a new reader to track and adds cmake complexity in my opinion.

Happy with this solution though, so marking as resolved.